networkx/utils/mapped_queue.py

```diff
@@ -43,7 +43,13 @@ def __lt__(self, other):
             return self.priority < other
         # assume comparing to another _HeapElement
         if self.priority == other_priority:
-            return self.element < other.element
+            try:
+                return self.element < other.element
+            except TypeError as err:
+                raise TypeError(
+                    f"{err}\nConsider using a tuple for the priority to avoid such comparisons: "
+                    f"{{k: (v, i) for i, (k,v) in enumerate(data.items())}}"
```

Comment:
I think this is needed to clearly mark what's actually part of the embedded expression... I'm not 100% sure though as I don't think I've ever run into this case before!

```suggestion
                    f"{ {k: (v, i) for i, (k,v) in enumerate(data.items())} }"
```